src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -177,5 +209,338 @@ private void ThrowIfDisposed()
                 throw new ObjectDisposedException(null);
             }
         }
+
+        private void InitializeMetricsClients()
+        {
+            if (_metricsReceiver == null)
+            {
+                _metricsReceiver = _messagingProvider.CreateMessageReceiver(_entityPath, _serviceBusAccount.ConnectionString);
+            }
+
+            if (_managementClient == null)
+            {
+                _managementClient = new ManagementClient(_serviceBusAccount.ConnectionString);
+            }
+        }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            if (_entityType == EntityType.Queue)
+            {
+                return await GetQueueMetricsAsync();
+            }
+            else
+            {
+                return await GetTopicMetricsAsync();
+            }
+        }
+
+        public async Task<ServiceBusTriggerMetrics> GetMetricsAsync()
+        {
+            if (_entityType == EntityType.Queue)
+            {
+                return await GetQueueMetricsAsync();
+            }
+            else
+            {
+                return await GetTopicMetricsAsync();
+            }
+        }
+
+        public async Task<ServiceBusTriggerMetrics> GetQueueMetricsAsync()
+        {
+            Message message = null;
+            QueueDescription queue;
+            QueueRuntimeInfo queueRuntime;
+            long deadLetterCount = 0, activeMessageCount = 0;
+            int partitionCount = 0;
+            TimeSpan eventQueueTime = TimeSpan.Zero;
+            bool isUnauthorized = false;
+
+            try
+            {
+                InitializeMetricsClients();
+
+                // Peek the first message in the queue without removing it from the queue
+                message = await _metricsReceiver.PeekAsync();
```

Comment:
Did you test it with sessions enabled entity? My understanding MessageReceiver does not work with sessions.